Shoko.Server/API/v3/Controllers/SeriesController.cs

```diff
@@ -182,6 +182,136 @@ public ActionResult<List<SeriesRelation>> GetAnidbRelationsBySeriesID([FromRoute
                 .ToList();
         }
 
+        #region Recommended For You
+
+        /// <summary>
+        /// Gets anidb recommendation for the user.
+        /// </summary>
+        /// <param name="pageSize">Limits the number of results per page. Set to 0 to disable the limit.</param>
+        /// <param name="page">Page number.</param>
+        /// <param name="showAll">If enabled will show recommendations across all the anidb available in Shoko, if disabled will only show for the user's collection.</param>
+        /// <param name="startDate">Start date to use if recommending for a watch period. Only setting the <paramref name="startDate"/> and not <paramref name="endDate"/> will result in using the watch history from the start date to the present date.</param>
+        /// <param name="endDate">End date to use if recommending for a watch period.</param>
+        /// <param name="approval">Minumum approval percentage for similar animes.</param>
+        /// <param name="recommendationType">Recommendation type for user reviews.</param>
+        /// <returns></returns>
+        [HttpGet("AniDB/RecommendedForYou")]
+        public ActionResult<List<Series.AniDBRecommendedForYou>> GetAnimeRecommendedForYou(
+            [FromQuery] int pageSize = 30,
+            [FromQuery] int page = 0,
+            [FromQuery] bool showAll = false,
+            [FromQuery] DateTime? startDate = null,
+            [FromQuery] DateTime? endDate = null,
+            [FromQuery] double? approval = null,
+            [FromQuery] Series.AniDBRecommendationType? recommendationType = null
+        )
+        {
+            if (approval.HasValue && (approval <= 0 || approval > 1))
+                approval = null;
+            if (startDate.HasValue && !endDate.HasValue)
+                endDate = DateTime.Now;
+            if (endDate.HasValue && !startDate.HasValue)
+                return BadRequest("Missing start date.");
+            if (startDate.HasValue && endDate.HasValue)
+            {
+                if (endDate.Value > DateTime.Now)
+                    return BadRequest("End date cannot be set into the future.");
+                if (startDate.Value > endDate.Value)
+                    return BadRequest("Start date cannot be newer than the end date.");
+            }

```

Comment:
Are we able to do this validation via attributes, to avoid boilerplate code?

Reply:
Will check.

Reply:
It seems like the `Range` attribute only supports static values, and there is no specific attribute for validating `DateTime` ranges, so I will prefer this boilerplate over the other solution of writing a custom attribute that would only be used once (resulting in even more boilerplate). And I still don't know if it loads the custom attribute into memory once the app starts (resulting in semi-static values) or if it checks the attribute every time the method is accessed.

Unless you (or any other members/users) know a better way then I will leave it there.